source/slang/slang-check-expr.cpp

```diff
@@ -3677,4 +3677,43 @@ namespace Slang
         }
     }
 
+    Expr* SemanticsExprVisitor::visitFuncTypeExpr(FuncTypeExpr* expr)
+    {
+        // The input and output to a function type must both be types
+        for(auto& t : expr->parameters)
+            t = CheckProperType(t);
+        expr->result = CheckProperType(expr->result);
+
```

Comment:
May need to return error expr here if expr->result is null.

Reply:
I can't see a similar case in the other visit functions, do you mean something like setting the whole type to `ErrorType` if one of the inputs or outputs is errortype?